srcpkgs/xbps-triggers/files/openjdk-profile

```diff
@@ -0,0 +1,39 @@
+#!/bin/sh
+#
+# Adds an entry to /etc/profile.d for setting $JAVA_HOME.
+#
+# Arguments:	$ACTION = [run/targets]
+#		$TARGET = [post-install/pre-remove]
+#		$PKGNAME
+#		$VERSION
+#		$UPDATE = [yes/no]
+#
+ACTION="$1"
+TARGET="$2"
+PKGNAME="$3"
+VERSION="$4"
+UPDATE="$5"
+
+jdkprofile=etc/profile.d/openjdk.sh
+
+case "$ACTION" in
+targets)
+	echo "post-install"
+	;;
+run)
+	[ -f "$jdkprofile" ] && exit 0
+
+	cat > "$jdkprofile" <<-'EOF'
+if _java_bin=$(readlink -e /usr/bin/java); then
+	_java_bin="${_java_bin%/bin/java}"
+	[ -d "${_java_bin}" ] && export JAVA_HOME="${JAVA_HOME:=${_java_bin}}"
+fi
+unset _java_bin
```

Comment:
The `-` in `<<-'EOF'` throws away leading tabs so the profile snippet it writes is a bit ugly. I recommend a single tab-indent for the *entire* block, including closing `EOF`, and then adding an extra 2 or 4 spaces on the lines that should be indented in the output.

Alternatively, just drop the `-` and the formatting will be preserved.

Reply:
done